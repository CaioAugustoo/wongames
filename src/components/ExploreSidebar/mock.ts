export default [
  {
    title: 'Preço',
    name: 'preco',
    type: 'checkbox',
    fields: [
      {
        label: 'Abaixo de R$50',
        name: 'abaixo-50'
      },
      {
        label: 'Abaixo de R$100',
        name: 'abaixo-100'
      },
      {
        label: 'Abaixo de R$150',
        name: 'abaixo-150'
      },
      {
        label: 'Abaixo de R$200',
        name: 'abaixo-200'
      },
      {
        label: 'Gratuito',
        name: 'gratuito'
      },
      {
        label: 'Em desconto',
        name: 'em-desconto'
      }
    ]
  },
  {
    title: 'Filtrar por',
    name: 'filtrar_por',
    type: 'radio',
    fields: [
      {
        label: 'Maior para menor',
        name: 'maior-para-menor'
      },
      {
        label: 'Menor para maior',
        name: 'menor-para-maior'
      }
    ]
  },
  {
    title: 'Sistema',
    name: 'sistema',
    type: 'checkbox',
    fields: [
      {
        label: 'Windows',
        name: 'windows'
      },
      {
        label: 'Linux',
        name: 'linux'
      },
      {
        label: 'MacOS',
        name: 'macos'
      }
    ]
  },
  {
    title: 'Gênero',
    name: 'genero',
    type: 'checkbox',
    fields: [
      {
        label: 'Ação',
        name: 'acao'
      },
      {
        label: 'Aventura',
        name: 'aventura'
      },
      {
        label: 'FPS',
        name: 'fps'
      },
      {
        label: 'MMORPG',
        name: 'mmorpg'
      },
      {
        label: 'RPG',
        name: 'rpg'
      },
      {
        label: 'Indie',
        name: 'indie'
      },
      {
        label: 'Tiros',
        name: 'tiros'
      },
      {
        label: 'Simulador',
        name: 'simulador'
      }
    ]
  }
]
