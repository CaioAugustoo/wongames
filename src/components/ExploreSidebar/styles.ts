import styled, { css, DefaultTheme } from 'styled-components'
import media from 'styled-media-query'
import { rgba } from 'polished'

import * as CheckboxStyles from 'components/Checkbox/styles'
import * as RadioStyles from 'components/Radio/styles'
import * as HeadingStyles from 'components/Heading/styles'

export type WrapperProps = {
  isOpen: boolean
}

export const IconWrapper = styled.div`
  cursor: pointer;
  width: 2.4rem;
  height: 2.4rem;

  ${media.greaterThan('medium')`
    display: none;
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    flex: 1;
    overflow-y: auto;
    padding: 0 ${theme.spacings.small};
    margin-top: 5.6rem;
    margin-bottom: 2rem;
    transition: transform ${theme.transition.default};

    ${media.greaterThan('medium')`
      overflow-y: initial;
      padding: 0;
      margin-top: 0;
      margin-bottom: 0;
    `}
  `}
`

export const Overlay = styled.div`
  ${({ theme }) => css`
    transition: opacity ${theme.transition.default};
    opacity: 0;
    position: absolute;
  `}
`

export const Items = styled.div`
  ${({ theme }) => css`
    & > div:not(:last-of-type) {
      margin-bottom: ${theme.spacings.xsmall};
    }
    & + div {
      border-top: 0.1rem solid ${rgba(theme.colors.gray, 0.2)};
      margin-top: ${theme.spacings.small};
      padding-top: ${theme.spacings.xsmall};
    }
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    box-shadow: 0 -0.2rem 0.4rem ${rgba(theme.colors.black, 0.2)};
    padding: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      display: none;
    `}
  `}
`

const wrapperModifiers = {
  open: (theme: DefaultTheme) => css`
    z-index: ${theme.layers.modal};
    position: fixed;
    width: 100%;
    height: 100vh;
    display: flex;
    flex-direction: column;
    top: 0;
    left: 0;

    ${Overlay} {
      z-index: ${theme.layers.modal};
      background-color: #fff;
      top: 0;
      left: 0;
      position: fixed;
      width: 100%;
      height: 100vh;
      opacity: 1;
    }

    ${Content} {
      margin-top: ${theme.spacings.medium};
      transform: translateY(0);
      overflow-y: scroll;
    }

    ${Content}, ${Footer}, ${IconWrapper} {
      z-index: ${theme.layers.modal};
    }

    ${HeadingStyles.Wrapper} {
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.xlarge};
      font-weight: ${theme.font.normal};
    }

    ${RadioStyles.Label},
    ${CheckboxStyles.Label} {
      color: ${theme.colors.black};
    }

    ${IconWrapper} {
      color: ${theme.colors.black};
      > svg {
        position: absolute;
        width: 30px;
        right: 0.8rem;
        top: 0.8rem;

        &:first-child {
          display: none;
        }
      }
    }
  `,

  close: (theme: DefaultTheme) => css`
    ${IconWrapper} {
      color: ${theme.colors.white};

      > svg:last-child {
        display: none;
      }
    }

    ${Content} {
      transform: translateY(3rem);
      height: 0;
    }

    ${Content}, ${Footer} {
      visibility: hidden;
      position: absolute;
      left: 0;
    }
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, isOpen }) => css`
    ${media.lessThan('medium')`
      ${!!isOpen && wrapperModifiers.open(theme)}
      ${!isOpen && wrapperModifiers.close(theme)}
    `}
  `}
`
