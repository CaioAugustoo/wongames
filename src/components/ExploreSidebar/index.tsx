import * as S from './styles'

const ExploreSidebar = () => (
  <S.Wrapper>
    <h1>ExploreSidebar</h1>
  </S.Wrapper>
)

export default ExploreSidebar
