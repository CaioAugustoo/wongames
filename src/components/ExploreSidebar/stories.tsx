import { Story, Meta } from '@storybook/react/types-6-0'
import ExploreSidebar from '.'

export default {
  title: 'ExploreSidebar',
  component: ExploreSidebar,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  }
} as Meta

export const Default: Story = () => <ExploreSidebar />
