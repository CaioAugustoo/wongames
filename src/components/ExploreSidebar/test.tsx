import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import ExploreSidebar from '.'
import items from './mock'

describe('<ExploreSidebar />', () => {
  it('should render headings', () => {
    renderWithTheme(<ExploreSidebar items={items} />)

    expect(screen.getByRole('heading', { name: /Preço/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Filtrar por/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Sistema/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /Gênero/i })).toBeInTheDocument()
  })

  it('should render inputs', () => {
    renderWithTheme(<ExploreSidebar items={items} />)

    expect(
      screen.getByRole('checkbox', { name: /Abaixo de R\$50/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('radio', { name: /Menor para maior/i })
    ).toBeInTheDocument()
  })

  it('should render filter button', () => {
    renderWithTheme(<ExploreSidebar items={items} />)

    expect(screen.getByRole('button', { name: /Filtrar/i })).toBeInTheDocument()
  })
})
