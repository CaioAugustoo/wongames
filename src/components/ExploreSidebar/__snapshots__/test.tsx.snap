// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExploreSidebar /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    ExploreSidebar
  </h1>
</main>
`;
